### PR TITLE
prov/tcp fix windows warnings

### PR DESCRIPTION
**Pull request by vkrishna**
Signed-off-by: Venkata Krishna Nimmagadda <venkata.krishna.nimmagadda@intel.com>

### CODE REVIEW

File: prov/tcp/src/tcpx_conn_mgr.c [hunk 1]
```diff
@@ -219,15 +219,15 @@ static int rx_cm_data(SOCKET fd, struct ofi_ctrl_hdr *hdr,
...
 {
 	struct ofi_ctrl_hdr hdr;
-	int ret;
+	ssize_t ret;
 
 	memset(&hdr, 0, sizeof(hdr));
 	hdr.version = OFI_CTRL_VERSION;
 	hdr.type = type;
-	hdr.seg_size = htons(poll_info->cm_data_sz);
+	hdr.seg_size = (uint16_t) htons(poll_info->cm_data_sz);
```
Comment (shefty) [inline prov/tcp/src/tcpx_conn_mgr.c]: What is windows doing that it needs this cast?  odd...

File: prov/tcp/src/tcpx_ep.c [hunk 2]
```diff
@@ -286,13 +287,13 @@ static int tcpx_setup_socket(SOCKET sock)
...
 
-	ret = setsockopt(sock, SOL_SOCKET, SO_REUSEADDR, &optval, sizeof(optval));
+	ret = setsockopt(sock, SOL_SOCKET, SO_REUSEADDR, (const char *)&optval, sizeof(optval));
 	if (ret) {
 		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,"setsockopt reuseaddr failed\n");
 		return ret;
 	}
 
-	ret = setsockopt(sock, IPPROTO_TCP, TCP_NODELAY, &optval, sizeof(optval));
+	ret = setsockopt(sock, IPPROTO_TCP, TCP_NODELAY, (const char *)&optval, sizeof(optval));
```
Comment (shefty) [inline prov/tcp/src/tcpx_ep.c]: Please check line length here and above and try to keep to around 80 characters

File: prov/tcp/src/tcpx_conn_mgr.c (hunk 1 above)
Comment (vkrishna) [reply prov/tcp/src/tcpx_conn_mgr.c]: That is a mistake. It should be
hdr.seg_size = htons((uint16_t) poll_info->cm_data_sz);

Comment (vkrishna) [reply prov/tcp/src/tcpx_conn_mgr.c]: There are still some warnings related to tcp. I will update this PR again.

**Comment by shefty** [comment]
Replaced by commit in PR #3817.

**Pull request closed without merging**